{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# most variable genes.r\n",
    "\n",
    "\n",
    "options(stringsAsFactors=FALSE) # for compatibile code between us\n",
    "\n",
    "library(tidyverse)\n",
    "\n",
    "\n",
    "getwd() <-\n",
    "  \"~/Documents/UCSC/Junior/Treehouse/Treehouse_OutlierRNASeq/\"\n",
    "\n",
    "setwd(paste0(getwd(), \"comp4.3_tert8.ckcc.outlier_results\"))\n",
    "\n",
    "up_outlier_files=list.files(, \"outlier_results_\")\n",
    "\n",
    "outlierResults<-lapply(up_outlier_files, function(x) {\n",
    "  read_tsv(x, col_types=cols()) %>%\n",
    "    add_column(sampleID=gsub(\"outlier_results_\", \"\", x))\n",
    "}) \t%>%\n",
    "  bind_rows()\n",
    "\n",
    "dfGeneMean <- outlierResults %>%\n",
    "\tgroup_by(Gene) %>%\n",
    "\tsummarize(mean = mean(sample))\n",
    "\n",
    "mean(dfGeneMean$mean)\n",
    "# the overall mean is 1.019 sample \n",
    "\n",
    "dfGeneVar <- outlierResults %>%\n",
    "\tgroup_by(Gene) %>%\n",
    "\tsummarize(variation = var(sample)) %>%\n",
    "\tarrange(desc(variation))\n",
    "\n",
    "mean(dfGeneVar$variation) # 0.56\n",
    "# so most genes differ from the norm on average by 0.56 \n",
    "\n",
    "sd(dfGeneVar$variation) # standard deviation = 1.15\n",
    "\n",
    "summary(dfGeneVar)\n",
    " #     Gene             variation        \n",
    " # Length:58581       Min.   : 0.000000  \n",
    " # Class :character   1st Qu.: 0.001579  \n",
    " # Mode  :character   Median : 0.069611  \n",
    " #                    Mean   : 0.565726  \n",
    " #                    3rd Qu.: 0.707318  \n",
    " #                    Max.   :28.579563  \n",
    "\n",
    "\n",
    "quantile(dfGeneVar$variation, 0.95)\n",
    "# > 95% of the data variates from the mean by 2.56\n",
    "\n",
    "\n",
    "\n",
    "geneList <- dfGeneVar %>% filter(variation > quantile(dfGeneVar$variation, 0.95))\n",
    "# get names of genes p95 of variation and up\n",
    "\n",
    "dfPercentile <- outlierResults %>%\n",
    "\tgroup_by(sampleID) %>%\n",
    "\tsummarize(p95 = quantile(sample, 0.95))\n",
    "\n",
    "dfSamples <-\n",
    "\t\toutlierResults %>% group_by(sampleID) %>% filter(Gene %in% geneList$Gene)\n",
    "\t# match names to all of their th01 th02 etc...\n",
    "\n",
    "dfSamples$TH01 <- grepl(pattern = 'TH01', dfSamples$sampleID)\n",
    "dfSamples$TH01 <- gsub('TRUE', 'blue',dfSamples$TH01)\n",
    "dfSamples$TH01 <- gsub('FALSE', 'red',dfSamples$TH01)\n",
    "\n",
    "\n",
    "\n",
    "sampleList <- dfSamples %>%\n",
    "\tselect(sampleID,sample) %>%\n",
    "\tgroup_by(sampleID) %>%\n",
    "\tsummarize()\n",
    "\n",
    "\n",
    "for (thisSample in sampleList$sampleID) {\n",
    "\tprint(thisSample)\n",
    "\tdfi <- dfSamples %>% filter(sampleID == thisSample)\n",
    "\tdfp <- dfPercentile %>% filter(sampleID == thisSample)\n",
    "\tdfc <- dfSamples %>% group_by(TH01) %>% filter(sampleID == thisSample) %>% summarize()\n",
    "\tprint(dfc[1][[1]])\n",
    "\tggplot(dfSamples %>% filter(sampleID == thisSample)) +\n",
    "\t\tgeom_histogram(aes(sample), binwidth = 0.1) +\n",
    "\t\tscale_fill_manual(values = c('red')) +\n",
    "\t\tggtitle(thisSample) +\n",
    "\t\txlab(\"log2(TPM+1)\") + ylab(\"Gene Expression\") +\n",
    "\t\tscale_x_continuous(limits = c(0,20)) +\n",
    "\t\tscale_y_continuous(limits = c(0,200)) +\n",
    "\t\tgeom_vline(xintercept = dfp$p95) +\n",
    "\t\tannotate(\n",
    "\t\t\t\t\t\"text\",\n",
    "\t\t\t\t\tx = round(dfp$p95,4)+4.3,\n",
    "\t\t\t\t\ty = 150,\n",
    "\t\t\t\t\tlabel = paste0(\n",
    "\t\t\t\t\t\t\"glp95: \",\n",
    "\t\t\t\t\t\tround(dfp$p95,4)\n",
    "\t\t\t\t\t)\n",
    "\t\t\t\t) \n",
    "\n",
    "\tmaxGene <- max(dfi$sample)\n",
    "\tmaxVarGene<-dfi[which.max(dfi$sample),]$Gene\n",
    "\tvariationOfMax <- round(maxGene- mean(dfi$sample),3) \n",
    "\tggsave(\n",
    "\t\tpaste0(\n",
    "\t\t\t\"pctl=\",\n",
    "\t\t\tformat(round(dfp$p95,4),nsmall = 4),\n",
    "\t\t\t\"-var=\",\n",
    "\t\t\tvariationOfMax,\n",
    "\t\t\t\"-\",\n",
    "\t\t\tthisSample,\n",
    "\t\t\t\".png\"\n",
    "\t\t),\n",
    "\t\tplot = p,\n",
    "\t\t\"png\",\n",
    "\t\tpaste0(getwd(), \"Batch-MostVariantGenesSorted-by-p95\")\n",
    "\t)\n",
    "\n",
    "}\n",
    "\n",
    "\n",
    "# saves plots and bumps for 22 sample files on the low end of the 95th percentile\n",
    "{\n",
    "\tnfpDF <-\n",
    "\t\tdfSamples %>% group_by(sampleID) %>% summarize(nfp = quantile(sample, 0.95))\n",
    "\t\n",
    "\tfifteenth = quantile(nfpDF$nfp, 0.15)\n",
    "\tworst15pctSamples <-\n",
    "\t\tnfpDF %>% filter(nfp < fifteenth) %>% arrange(desc(nfp))\n",
    "\t\n",
    "\tthisSample <- NULL\n",
    "\torder <- 0\n",
    "\tsumBump <- 0\n",
    "\tcountBump <- 0 \n",
    "\taverageBump <- 0 \n",
    "\n",
    "\tx <- list()\n",
    "\tfor (thisSample in worst15pctSamples$sampleID) {\n",
    "\t\tprint(thisSample)\n",
    "\t\tdf <-\n",
    "\t\t\tdfSamples %>% filter(sampleID == thisSample) %>% filter(sample > 1.8)\n",
    "\t\tdfn <- count(df, sample=round(sample,1))\n",
    "\t\tdfn$index <- seq(1,length(dfn$n))\n",
    "\n",
    "\t\tmaxGene <- max(df$sample)\n",
    "\t\tmaxVarGene<-df[which.max(df$sample),]$Gene\n",
    "\t\tvariationOfMax <- round(maxGene- mean(df$sample),3) \n",
    "\n",
    "\t\torder = order + 1\n",
    "\t\tp <- ggplot(dfSamples %>% filter(sampleID == thisSample)) +\n",
    "\t\t\tgeom_histogram(aes(sample), binwidth = 0.1) +\n",
    "\t\t\tggtitle(paste0(thisSample,\"\\n maxGene: \", maxVarGene)) +\n",
    "\t\t\tscale_x_continuous(limits = c(0,15))+\n",
    "\t\t\tscale_y_continuous(limits = c(0,125))+\n",
    "\t\t\txlab(\"log2(TPM+1)\") + ylab(\"Gene Expression\")\n",
    "\n",
    "\t\tif(dfn[which.max(dfn$n),]$sample > 2.1) {\n",
    "\t\t\tp = p + annotate(\n",
    "\t\t\t\t\t\"text\",\n",
    "\t\t\t\t\tx = dfn[which.max(dfn$n),]$sample+3,\n",
    "\t\t\t\t\ty = 1000,\n",
    "\t\t\t\t\tlabel = paste0(\n",
    "\t\t\t\t\t\t\"bump: \",\n",
    "\t\t\t\t\t\tdfn[which.max(dfn$n),]$sample\n",
    "\n",
    "\t\t\t\t\t)\n",
    "\t\t\t\t) + geom_vline(xintercept = dfn[which.max(dfn$n),]$sample)\n",
    "\t\t\tsumBump = sumBump + dfn[which.max(dfn$n),]$sample\n",
    "\t\t\tcountBump = countBump + 1\n",
    "\t\t}\n",
    "\t\n",
    "\n",
    "\t\tggsave(\n",
    "\t\t\tpaste0(\n",
    "\t\t\t\torder + 100,\n",
    "\t\t\t\t\"_\",\n",
    "\t\t\t\tround(worst15pctSamples[[2]][order], digits = 3),\n",
    "\t\t\t\t\"_\",\n",
    "\t\t\t\tthisSample,\n",
    "\t\t\t\t\".png\"\n",
    "\t\t\t),\n",
    "\t\t\tplot = p,\n",
    "\t\t\t\"png\",\n",
    "\t\t\tpaste0(getwd(), \"BatchPlotsMostVar-Below-p15\")\n",
    "\t\t)\n",
    "\t\t\n",
    "\t}\n",
    "\taverageBump = sumBump / countBump\n",
    "\tprint(averageBump)\n",
    "}\n",
    "\n",
    "\n",
    "# saves plots and bumps for 22 sample files on the high end of the 95th percentile\n",
    "{\n",
    "\tnfpDF <-\n",
    "\t\tdfSamples %>% group_by(sampleID) %>% summarize(nfp = quantile(sample, 0.95))\n",
    "\t\n",
    "\teightyFive = quantile(nfpDF$nfp, 0.85)\n",
    "\tbest85pctSamples <-\n",
    "\t\tnfpDF %>% filter(nfp > eightyFive) %>% arrange(desc(nfp))\n",
    "\t\n",
    "\tthisSample <- NULL\n",
    "\torder <- 0\n",
    "\tsumBump <- 0\n",
    "\tcountBump <- 0 \n",
    "\taverageBump <- 0 \n",
    "\n",
    "\tx <- list()\n",
    "\tfor (thisSample in best95pctSamples$sampleID) {\n",
    "\t\tprint(thisSample)\n",
    "\t\tdf <-\n",
    "\t\t\tdfSamples %>% filter(sampleID == thisSample) %>% filter(sample > 1.8)\n",
    "\t\tdfn <- count(df, sample=round(sample,1))\n",
    "\t\tdfn$index <- seq(1,length(dfn$n))\n",
    "\n",
    "\t\tmaxGene <- max(df$sample)\n",
    "\t\tmaxVarGene<-df[which.max(df$sample),]$Gene\n",
    "\t\tvariationOfMax <- round(maxGene- mean(df$sample),3) \n",
    "\n",
    "\t\torder = order + 1\n",
    "\t\tp <- ggplot(dfSamples %>% filter(sampleID == thisSample)) +\n",
    "\t\t\tgeom_histogram(aes(sample), binwidth = 0.1) +\n",
    "\t\t\tggtitle(paste0(thisSample,\"> p85\\n maxGene: \", maxVarGene)) +\n",
    "\t\t\tscale_x_continuous(limits = c(0,20))+\n",
    "\t\t\tscale_y_continuous(limits = c(0,300))+\n",
    "\t\t\txlab(\"log2(TPM+1)\") + ylab(\"Gene Expression\")+\n",
    "\t\t\tgeom_vline(xintercept=)\n",
    "\n",
    "\t\tif(dfn[which.max(dfn$n),]$sample > 2.5) {\n",
    "\t\t\tp = p + annotate(\n",
    "\t\t\t\t\t\"text\",\n",
    "\t\t\t\t\tx = dfn[which.max(dfn$n),]$sample+3,\n",
    "\t\t\t\t\ty = 1000,\n",
    "\t\t\t\t\tlabel = paste0(\n",
    "\t\t\t\t\t\t\"bump: \",\n",
    "\t\t\t\t\t\tdfn[which.max(dfn$n),]$sample\n",
    "\n",
    "\t\t\t\t\t)\n",
    "\t\t\t\t) + geom_vline(xintercept = dfn[which.max(dfn$n),]$sample)\n",
    "\t\t\tsumBump = sumBump + dfn[which.max(dfn$n),]$sample\n",
    "\t\t\tcountBump = countBump + 1\n",
    "\t\t}\n",
    "\t\n",
    "\n",
    "\t\tggsave(\n",
    "\t\t\tpaste0(\n",
    "\t\t\t\torder + 100,\n",
    "\t\t\t\t\"_\",\n",
    "\t\t\t\tround(best95pctSamples[[2]][order], digits = 3),\n",
    "\t\t\t\t\"_\",\n",
    "\t\t\t\tthisSample,\n",
    "\t\t\t\t\".png\"\n",
    "\t\t\t),\n",
    "\t\t\tplot = p,\n",
    "\t\t\t\"png\",\n",
    "\t\t\tpaste0(getwd(), \"BatchPlotsMostVar-Above-p95\")\n",
    "\t\t)\n",
    "\t\t\n",
    "\t}\n",
    "\taverageBump = sumBump / countBump\n",
    "\tprint(averageBump)\n",
    "}\n",
    "nfpDF <-\n",
    "\t\tdfSamples %>% group_by(sampleID) %>% summarize(nfp = quantile(sample, 0.95))\n",
    "\t\n",
    "eightyFive = quantile(nfpDF$nfp, 0.85)\n",
    "best85pctSamples <-\n",
    "\tnfpDF %>% filter(nfp > eightyFive) %>% arrange(desc(nfp))\n",
    "fifteenth = quantile(nfpDF$nfp, 0.15)\n",
    "worst15pctSamples <-\n",
    "\tnfpDF %>% filter(nfp < fifteenth) %>% arrange(desc(nfp))\n",
    "\n",
    "maxGene <- max(dfSamples$sample)\n",
    "maxVarGene<-dfSamples[which.max(dfSamples$sample),]$Gene\n",
    "variationOfMax <- round(maxGene- mean(dfSamples$sample),3) \n",
    "\n",
    "pctl <- data.frame(outlierResults %>% group_by(sampleID) %>% summarize(global95 = quantile(sample, 0.95)))\n",
    "\n",
    "dfSamples <- left_join(dfSamples, pctl, by=\"sampleID\")\n",
    "\n",
    "dfSamples %>% arrange((global95))\n",
    "\n",
    "dfSamples$TH01 <- grepl(\"TH01\", dfSamples$sampleID)\n",
    "\n",
    "facetBigPlot <- ggplot(dfSamples, aes(sample, fill= TH01)) + geom_histogram(binwidth=0.1) +\n",
    "\tggtitle(paste0(\"Samples All | maxGene: \", maxVarGene, \" | Distance From Mean: \", variationOfMax)) +\n",
    "\txlab(\"log2(TPM+1)\") + ylab(\"Gene Expression\")+\n",
    "\tscale_x_continuous(limits = c(0,20)) +\n",
    "\tscale_y_continuous(limits = c(0,100)) +\n",
    "\tscale_fill_brewer(palette = \"Set1\") +\n",
    "\tfacet_wrap(~ global95)\n",
    "\n",
    "\n",
    "ggsave(filename = \"facetWrapColored3.png\", facetBigPlot,\n",
    "       width = 20, height = 20, dpi = 150, units = \"in\", device='png', paste0(getwd()))\n",
    "\n",
    "ggplot(dfSamples %>% filter(sampleID == best85pctSamples$sampleID), aes(sample)) +\n",
    "\tgeom_histogram(binwidth=0.1) +\n",
    "\tggtitle(paste0(\"Highest 22 p95s | maxVarGene: \", maxVarGene, \" | Distance From Mean: \", variationOfMax)) +\n",
    "\txlab(\"log2(TPM+1)\") + ylab(\"Gene Expression\")+\t\n",
    "\tscale_x_continuous(limits = c(0,17)) +\n",
    "\tscale_y_continuous(limits = c(0,20)) +\n",
    "\tfacet_wrap(~ sampleID)\n",
    "\n",
    "dfBadSamples <- dfSamples %>% filter(sampleID == worst15pctSamples$sampleID)\n",
    "maxGene <- max(dfBadSamples$sample)\n",
    "maxVarGene<-dfBadSamples[which.max(dfBadSamples$sample),]$Gene\n",
    "variationOfMax <- round(maxGene- mean(dfBadSamples$sample),3) \n",
    "\n",
    "ggplot(dfBadSamples, aes(sample)) +\n",
    "\tgeom_histogram(binwidth=0.1) +\n",
    "\tggtitle(paste0(\"Lowest 22 p95s | maxVarGene: \", maxVarGene, \" | Distance From Mean: \", variationOfMax)) +\n",
    "\txlab(\"log2(TPM+1)\") + ylab(\"Gene Expression\")+\n",
    "\tscale_x_continuous(limits = c(0,17)) +\n",
    "\tscale_y_continuous(limits = c(0,20)) +\n",
    "\tfacet_wrap(~ sampleID)\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.3.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
