{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "\n",
    "options(stringsAsFactors = FALSE) # for compatibile code between us\n",
    "\n",
    "library(tidyverse)\n",
    "\n",
    "\n",
    "liamsWorkingDir <-\n",
    "  \"~/Documents/UCSC/Junior/Treehouse/Treehouse_OutlierRNASeq/\"\n",
    "setwd(liamsWorkingDir)\n",
    "\n",
    "umend_file = list.files(, \"ckccSampleUMEND_reads\")\n",
    "\n",
    "umendResults <- lapply(umend_file, function(x) {\n",
    "  read_tsv(x, col_types = cols()) %>%\n",
    "    add_column(sampleID = gsub(\"outlier_results_\", \"\", x))\n",
    "}) \t%>%\n",
    "  bind_rows() %>%\n",
    "  arrange(th_sampleid)\n",
    "\n",
    "\n",
    "# ''' plot 95 percentile vs UMEND reads per sample '''\n",
    "\n",
    "setwd(paste0(liamsWorkingDir, \"comp4.3_tert8.ckcc.outlier_results\"))\n",
    "\n",
    "up_outlier_files = list.files(, \"outlier_results_\")\n",
    "\n",
    "outlierResults <- lapply(up_outlier_files, function(x) {\n",
    "  read_tsv(x, col_types = cols()) %>%\n",
    "    add_column(sampleID = gsub(\"outlier_results_\", \"\", x))\n",
    "}) \t%>%\n",
    "  bind_rows()\n",
    "\n",
    "dfNotZerosOrNotZeros <- outlierResults %>%\n",
    "  select(sampleID, sample) %>%\n",
    "  group_by(sampleID) %>%\n",
    "  count(sample == 0) \n",
    "\n",
    "dfNotZeros <- dfNotZerosOrNotZeros %>%\n",
    "  group_by(sampleID) %>%\n",
    "  filter(`sample == 0` == F)\n",
    "\n",
    "\n",
    "UMENDp95DF <- outlierResults %>%\n",
    "  group_by(sampleID) %>%\n",
    "  summarise(p95 = quantile(sample, c(0.95))) %>%\n",
    "  arrange(sampleID) %>%\n",
    "  add_column(numberOfExpressed = dfNotZeros$n) %>%\n",
    "  filter(sampleID %in% umendResults$th_sampleid) %>%\n",
    "  add_column(rawUMEND = umendResults$umendCountRaw, \n",
    "    umendID = umendResults$th_sampleid, \n",
    "    disease = umendResults$disease)\n",
    "\n",
    "\n",
    "\n",
    "UMENDp95DF$Sample = grepl(\"TH01\", UMENDp95DF$umendID)\n",
    "\n",
    "UMENDp95DFTH01 <- UMENDp95DF %>% filter(Sample == T)\n",
    "UMENDp95DFOther <- UMENDp95DF %>% filter(Sample == F)\n",
    "\n",
    "UMENDp95DF$Sample <- gsub(\"TRUE\", \"TH01_[Ribo]...\",UMENDp95DF$Sample)\n",
    "UMENDp95DF$Sample <- gsub(\"FALSE\", \"Not TH01_[PolyA]...\",UMENDp95DF$Sample)\n",
    "\n",
    "UMENDp95DF$shortSampleID <- gsub('[_][0-9S]+','',UMENDp95DF$sampleID)\n",
    "i<-1\n",
    "corlist<-list(0)\n",
    "for(sampleCenter in UMENDp95DF$shortSampleID){\n",
    "  print(sampleCenter)\n",
    "  dfab <- UMENDp95DF %>% filter(shortSampleID == sampleCenter)\n",
    "  corlist[[i]] <- round(cor(dfab$rawUMEND, dfab$p95),4)\n",
    "  i <- i+ 1\n",
    "}\n",
    "\n",
    "correlations<-data.frame(sampleID = unique(UMENDp95DF$shortSampleID))\n",
    "correlations$cor <- unique(corlist)\n",
    "resultsCorrelations <- paste((paste0(correlations$sampleID, \": \", correlations$cor, \"; \\n\")), collapse = '')\n",
    "\n",
    "\n",
    "ggplot(UMENDp95DF, aes(p95, rawUMEND, color = shortSampleID)) + geom_point() +\n",
    "  ylab(\"Raw UMEND Count\") + xlab(\"95th Percentile Per Sample\") + \n",
    "  geom_smooth(method = 'lm', level = 0.95, se = TRUE) +\n",
    "  ggtitle(\"95th Percentile vs. UMEND Reads Per Sample\") + \n",
    "  annotate(\n",
    "    \"text\",\n",
    "    x = 3,\n",
    "    y = 7e+07,\n",
    "    label = paste0(\"correlations:\\n\", resultsCorrelations)\n",
    "      \n",
    "    )\n",
    "  )\n",
    "\n",
    "ggplot(UMENDp95DF, aes(p95, numberOfExpressed/1000, color = disease)) + geom_point() +\n",
    "  ylab(\"Number of Expressed Genes (thousands)\") + xlab(\"95th Percentile Per Sample\") + \n",
    "  geom_smooth(method = 'lm', level = 0.95, se = FALSE) +\n",
    "  ggtitle(\"95th Percentile vs. Number of Expressed Genes Per Sample\")\n",
    "\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.3.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
