{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# ==================================================================================================\n",
    "# \n",
    "# \t\tALL SCATTER PLOTS FROM PRESENTATION\n",
    "# \n",
    "# ==================================================================================================\n",
    "# LOAD DATA FROM comp4.3_tert8.ckcc.outlier_results\n",
    "# ==================================================================================================\n",
    "options(stringsAsFactors=FALSE) # for compatibile code between us\n",
    "\n",
    "library(tidyverse)\n",
    "\n",
    "\n",
    "liamsWorkingDir <-\n",
    "  \"~/Documents/UCSC/Junior/Treehouse/Treehouse_OutlierRNASeq/\"\n",
    "\n",
    "setwd(paste0(liamsWorkingDir, \"comp4.3_tert8.ckcc.outlier_results\"))\n",
    "\n",
    "up_outlier_files=list.files(, \"outlier_results_\")\n",
    "\n",
    "outlierResults<-lapply(up_outlier_files, function(x) {\n",
    "  read_tsv(x, col_types=cols()) %>%\n",
    "    add_column(sampleID=gsub(\"outlier_results_\", \"\", x))\n",
    "}) \t%>%\n",
    "  bind_rows()\n",
    "\n",
    "# ==================================================================================================\n",
    "# Number of Expressed Genes vs. p95\n",
    "# ==================================================================================================\n",
    "dfNotZerosOrNotZeros <- outlierResults %>%\n",
    "  select(sampleID, sample) %>%\n",
    "  group_by(sampleID) %>%\n",
    "  count(sample == 0) \n",
    "\n",
    "dfNotZeros <- dfNotZerosOrNotZeros %>%\n",
    "  group_by(sampleID) %>%\n",
    "  filter(`sample == 0` == F)\n",
    "\n",
    "p95df <- outlierResults %>% group_by(sampleID) %>% summarize(p95 = quantile(sample, 0.95))\n",
    "\n",
    "dfNotZeros$p95 = p95df$p95\n",
    "\n",
    "dfNotZeros$TH01 = grepl(\"TH01\", p95df$sampleID)\n",
    "\n",
    "dfTH01s <- dfNotZeros %>% filter(TH01 == T)\n",
    "dfNotTH01s <- dfNotZeros %>% filter(TH01 == F)\n",
    "\n",
    "dfNotZeros$TH01 <- gsub(\"TRUE\", \"RiboMinus\",dfNotZeros$TH01)\n",
    "dfNotZeros$TH01 <- gsub(\"FALSE\", \"PolyA\",dfNotZeros$TH01)\n",
    "\n",
    "\n",
    "dfScatter <- dfNotZeros \n",
    "colnames(dfScatter)[which(names(dfScatter) == \"TH01\")] <- \"Sample_Name\"\n",
    "dfScatter$Has_Bump <- allSamples[[3]]\n",
    "dfScatter$Bump_Value  <- allSamples[[4]]\n",
    "ggplot(dfScatter,aes(n/1000,p95, color = Sample_Name)) + \n",
    "  scale_colour_brewer(type = \"seq\", palette = \"Set1\", direction = 1)+\n",
    "  scale_fill_brewer(type = \"seq\", palette = \"Set1\", direction = 1)+\n",
    "  geom_point(aes(shape=Has_Bump)) +\n",
    "  scale_size_continuous(range = c(1,6))+\n",
    "  xlab('Expressed Genes (Thousands)') + ylab('95th Percentile per Sample') +\n",
    "  ggtitle('Each Sample\\'s Count of Expressed Genes and its 95th Percentile') +\n",
    "  geom_smooth(method = 'lm')+\n",
    "  annotate(\n",
    "    \"text\",\n",
    "    x = 30,\n",
    "    y = 1 ,\n",
    "    label = paste0(\n",
    "      \"correlation TH01: \",\n",
    "      round(cor(dfTH01s$n,dfTH01s$p95),3),\n",
    "      \"\\ncorrelation Not TH01: \",\n",
    "      round(cor(dfNotTH01s$n,dfNotTH01s$p95),3)\n",
    "      \n",
    "    )\n",
    "  )\n",
    "\n",
    "dfScatter$shortSampleID <- gsub('[_][0-9S]+','',dfScatter$sampleID)\n",
    "i<-1\n",
    "corlist<-list(0)\n",
    "for(sampleCenter in dfScatter$shortSampleID){\n",
    "\tprint(sampleCenter)\n",
    "\tdfab <- dfScatter %>% filter(shortSampleID == sampleCenter)\n",
    "\tcorlist[[i]] <- round(cor(dfab$n, dfab$p95),4)\n",
    "\ti <- i+ 1\n",
    "}\n",
    "\n",
    "correlations<-data.frame(sampleID = unique(dfScatter$shortSampleID))\n",
    "correlations$cor <- unique(corlist)\n",
    "\n",
    "for (i in seq(1,4)){\n",
    "\tpaste0()\n",
    "}\n",
    "\n",
    "\n",
    "resultsCorrelations <- paste((paste0(correlations$sampleID, \": \", correlations$cor, \"; \\n\")), collapse = '')\n",
    "ggplot(dfScatter,aes(n/1000,p95, color = shortSampleID)) + \n",
    "  scale_colour_brewer(palette = \"Set1\")+\n",
    "  scale_fill_brewer(palette = \"Set1\")+\n",
    "  geom_point() +\n",
    "  scale_size_continuous(range = c(1,6))+\n",
    "  geom_smooth(method = 'lm')+\n",
    "  annotate(\n",
    "    \"text\",\n",
    "    x = 30,\n",
    "    y = 2.5 ,\n",
    "    label = paste0(\"correlations\\n\",resultsCorrelations)\n",
    "    )+\n",
    "  xlab('Expressed Genes (Thousands)') + ylab('95th Percentile per Sample') +\n",
    "  ggtitle('Each Sample\\'s Count of Expressed Genes and its 95th Percentile') \n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# ==================================================================================================\n",
    "# UMEND count vs p95\n",
    "# ==================================================================================================\n",
    "# LOAD DATA FROM ckccSampleUMEND_reads and comp4.3_tert8.ckcc.outlier_results\n",
    "# ==================================================================================================\n",
    "\n",
    "umend_file = list.files(, \"ckccSampleUMEND_reads\")\n",
    "\n",
    "umendResults <- lapply(umend_file, function(x) {\n",
    "  read_tsv(x, col_types = cols()) %>%\n",
    "    add_column(sampleID = gsub(\"outlier_results_\", \"\", x))\n",
    "}) \t%>%\n",
    "  bind_rows() %>%\n",
    "  arrange(th_sampleid)\n",
    "\n",
    "# ''' plot 95 percentile vs UMEND reads per sample '''\n",
    "\n",
    "setwd(paste0(liamsWorkingDir, \"comp4.3_tert8.ckcc.outlier_results\"))\n",
    "\n",
    "up_outlier_files = list.files(, \"outlier_results_\")\n",
    "\n",
    "outlierResults <- lapply(up_outlier_files, function(x) {\n",
    "  read_tsv(x, col_types = cols()) %>%\n",
    "    add_column(sampleID = gsub(\"outlier_results_\", \"\", x))\n",
    "}) \t%>%\n",
    "  bind_rows()\n",
    "\n",
    "\n",
    "\n",
    "UMENDp95DF <- outlierResults %>%\n",
    "  group_by(sampleID) %>%\n",
    "  summarise(p95 = quantile(sample, c(0.95))) %>%\n",
    "  arrange(sampleID) %>%\n",
    "  filter(sampleID %in% umendResults$th_sampleid) %>%\n",
    "  add_column(rawUMEND = umendResults$umendCountRaw, umendID = umendResults$th_sampleid)\n",
    "\n",
    "\n",
    "\n",
    "UMENDp95DF$Sample = grepl(\"TH01\", UMENDp95DF$umendID)\n",
    "\n",
    "UMENDp95DFTH01 <- UMENDp95DF %>% filter(Sample == T)\n",
    "UMENDp95DFOther <- UMENDp95DF %>% filter(Sample == F)\n",
    "\n",
    "UMENDp95DF$Sample <- gsub(\"TRUE\", \"TH01_[Ribo]...\",UMENDp95DF$Sample)\n",
    "UMENDp95DF$Sample <- gsub(\"FALSE\", \"Not TH01_[PolyA]...\",UMENDp95DF$Sample)\n",
    "\n",
    "UMENDp95DF$shortSampleID <- gsub('[_][0-9S]+','',UMENDp95DF$sampleID)\n",
    "i<-1\n",
    "corlist<-list(0)\n",
    "for(sampleCenter in UMENDp95DF$shortSampleID){\n",
    "  print(sampleCenter)\n",
    "  dfab <- UMENDp95DF %>% filter(shortSampleID == sampleCenter)\n",
    "  corlist[[i]] <- round(cor(dfab$rawUMEND, dfab$p95),4)\n",
    "  i <- i+ 1\n",
    "}\n",
    "\n",
    "correlations<-data.frame(sampleID = unique(UMENDp95DF$shortSampleID))\n",
    "correlations$cor <- unique(corlist)\n",
    "resultsCorrelations <- paste((paste0(correlations$sampleID, \": \", correlations$cor, \"; \\n\")), collapse = '')\n",
    "\n",
    "\n",
    "ggplot(UMENDp95DF, aes(p95, rawUMEND, color = shortSampleID)) + geom_point() +\n",
    "  ylab(\"Raw UMEND Count\") + xlab(\"95th Percentile Per Sample\") + \n",
    "  geom_smooth(method = 'lm', level = 0.95, se = TRUE) +\n",
    "  ggtitle(\"95th Percentile vs. UMEND Reads Per Sample\") + \n",
    "  annotate(\n",
    "    \"text\",\n",
    "    x = 3,\n",
    "    y = 7e+07,\n",
    "    label = paste0(\"correlations:\\n\", resultsCorrelations)\n",
    "      \n",
    "    )\n",
    "  )\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# ==================================================================================================\n",
    "# TPM vs. p95\n",
    "# ==================================================================================================\n",
    "options(stringsAsFactors=FALSE) # for compatibile code between us\n",
    "\n",
    "library(tidyverse)\n",
    "library(gridExtra) # easy for putting graphs onto the same page (just use ggarrange(graph1, graph2, ncol = # of display\n",
    "# columns, nrow = #row))\n",
    "\n",
    "\n",
    "setwd(\"~/Documents/UCSC/Junior/Treehouse/Treehouse_OutlierRNASeq/ckcc_rsem_genes_results\")\n",
    "\n",
    "sample_file_list=list.files(, \"TH0\")\n",
    "\n",
    "rawTPMDf<-lapply(sample_file_list, function(x) {\n",
    "\t\tread_tsv(x, col_types=cols()) %>%\n",
    "\t\tadd_column(sampleID=gsub(\"_rsem_genes\", \"\", x))\n",
    "\t\t})  %>%\n",
    "\t\tbind_rows()\n",
    "\n",
    "percentileOfEachTPMSampleDf <- rawTPMDf %>%\n",
    "\tgroup_by(sampleID) %>% \n",
    "\tsummarize(p95 = quantile(TPM, 0.95))\n",
    "\n",
    "dfNotZerosOrNotZeros <- rawTPMDf %>%\n",
    "\tselect(sampleID, TPM) %>%\n",
    "\tgroup_by(sampleID) %>%\n",
    "\tcount(TPM == 0) \n",
    "\n",
    "dfNotZeros <- dfNotZerosOrNotZeros %>%\n",
    "\tgroup_by(sampleID) %>%\n",
    "\tfilter(`TPM == 0` == F)\n",
    "\n",
    "\n",
    "dfNotZeros$p95<-percentileOfEachTPMSampleDf$p95\n",
    "dfNotZeros$zeroCount <- dfNotZeros$n\n",
    "\n",
    "dfScatter <- dfNotZeros \n",
    "ggplot(dfScatter, aes(p95, zeroCount/1000)) + \n",
    "\tscale_colour_brewer(type = \"seq\", palette = \"Set1\", direction = 1)+\n",
    "\tscale_fill_brewer(type = \"seq\", palette = \"Set1\", direction = 1)+\n",
    "\tgeom_point() +\n",
    "\tylab('Expressed Genes (Thousands)') + xlab('95th Percentile per Sample (TPM)') +\n",
    "\tggtitle('Each Sample\\'s Count of Expressed Genes and its 95th Percentile (TPM)') +\n",
    "\tgeom_smooth(method = 'lm')+\n",
    "\tannotate(\n",
    "\t\"text\",\n",
    "\tx = 30,\n",
    "\ty = 20 ,\n",
    "\tlabel = paste0(\n",
    "\t\t\"correlation : \",\n",
    "\t\tround(cor(dfScatter$n,dfScatter$p95),3)\n",
    "\t\t\n",
    "\t)\n",
    "\t)\n",
    "dfScatter$shortSampleID <- gsub('.results','',dfScatter$sampleID)\n",
    "\n",
    "dfScatter$shortSampleID <- gsub('[_][0-9S]+','',dfScatter$shortSampleID)\n",
    "dfScatter$Method <- gsub('[TH02-9]+[^TH01]', 'PolyA', dfScatter$shortSampleID)\n",
    "dfScatter$Method <- gsub('TH01', 'RiboD', dfScatter$Method)\n",
    "i<-1\n",
    "corlist<-list(0)\n",
    "for(sampleCenter in dfScatter$shortSampleID){\n",
    "  print(sampleCenter)\n",
    "  dfab <- dfScatter %>% filter(shortSampleID == sampleCenter)\n",
    "  corlist[[i]] <- round(cor(dfab$n, dfab$p95),4)\n",
    "  i <- i+ 1\n",
    "}\n",
    "\n",
    "correlations<-data.frame(sampleID = unique(dfScatter$shortSampleID))\n",
    "correlations$cor <- unique(corlist)\n",
    "resultsCorrelations <- paste((paste0(correlations$sampleID, \": \", correlations$cor, \"; \\n\")), collapse = '')\n",
    "\n",
    "i<-1\n",
    "corlistM<-list(0)\n",
    "for(method in unique(dfScatter$Method)){\n",
    "  print(method)\n",
    "  dfab <- dfScatter %>% filter(Method == method)\n",
    "  corlistM[[i]] <- round(cor(dfab$n, dfab$p95),4)\n",
    "  i <- i+ 1\n",
    "}\n",
    "\n",
    "correlationsM<-data.frame(method = unique(dfScatter$Method))\n",
    "correlationsM$corMethod <- corlistM\n",
    "resultsCorrelationsM <- paste((paste0(correlationsM$method, \": \", correlationsM$corMethod, \"; \\n\")), collapse = '')\n",
    "\n",
    "\n",
    "ggplot(dfScatter, aes(p95, zeroCount/1000, color=shortSampleID)) + \n",
    "\tscale_colour_brewer(type = \"seq\", palette = \"Set1\", direction = 1)+\n",
    "\tscale_fill_brewer(type = \"seq\", palette = \"Set1\", direction = 1)+\n",
    "\tgeom_point() +\n",
    "\tylab('Expressed Genes (Thousands)') + xlab('95th Percentile per Sample (TPM)') +\n",
    "\tggtitle('Each Sample\\'s Count of Expressed Genes and its 95th Percentile (TPM)') +\n",
    "\tgeom_smooth(method = 'lm')+\n",
    "\tannotate(\"text\",x = 15,y = 35 ,label = paste0(\t\"correlation : \\n\",\tresultsCorrelations\t))\n",
    "ggplot(dfScatter, aes(p95, zeroCount/1000, color=Method)) + \n",
    "\tscale_colour_brewer(type = \"seq\", palette = \"Set1\", direction = 1)+\n",
    "\tscale_fill_brewer(type = \"seq\", palette = \"Set1\", direction = 1)+\n",
    "\tgeom_point() +\n",
    "\tylab('Expressed Genes (Thousands)') + xlab('95th Percentile per Sample (TPM)') +\n",
    "\tggtitle('Each Sample\\'s Count of Expressed Genes and its 95th Percentile (TPM)') +\n",
    "\tgeom_smooth(method = 'lm')+\n",
    "\tannotate(\"text\",x = 15,y = 35 ,label = paste0(\t\"correlation : \\n\",\tresultsCorrelationsM\t))\n",
    "\n",
    "\n",
    "# ==================================================================================================\n",
    "# UMEND vs. Expressed Genes per Sample\n",
    "# ==================================================================================================\n",
    "# plot # Expressed genes vs UMEND reads per sample;\n",
    "options(stringsAsFactors = FALSE) # for compatibile code between us\n",
    "\n",
    "library(tidyverse)\n",
    "\n",
    "\n",
    "liamsWorkingDir <-\n",
    "  \"~/Documents/UCSC/Junior/Treehouse/Treehouse_OutlierRNASeq/\"\n",
    "setwd(liamsWorkingDir)\n",
    "\n",
    "umend_file = list.files(, \"ckccSampleUMEND_reads\")\n",
    "\n",
    "umendResults <- lapply(umend_file, function(x) {\n",
    "  read_tsv(x, col_types = cols()) %>%\n",
    "    add_column(sampleID = gsub(\"outlier_results_\", \"\", x))\n",
    "}) \t%>%\n",
    "  bind_rows() %>%\n",
    "  arrange(th_sampleid)\n",
    "\n",
    "\n",
    "# plot # Expressed genes vs UMEND reads per sample;\n",
    "\n",
    "\n",
    "setwd(paste0(liamsWorkingDir, \"comp4.3_tert8.ckcc.outlier_results\"))\n",
    "\n",
    "\n",
    "up_outlier_files = list.files(, \"outlier_results_\")\n",
    "\n",
    "outlierResults <- lapply(up_outlier_files, function(x) {\n",
    "  read_tsv(x, col_types = cols()) %>%\n",
    "    add_column(sampleID = gsub(\"outlier_results_\", \"\", x))\n",
    "}) \t%>%\n",
    "  bind_rows()\n",
    "\n",
    "dfNonZeros <- outlierResults %>%\n",
    "  select(sampleID, sample) %>%\n",
    "  group_by(sampleID) %>%\n",
    "  count(sample == 0) %>%\n",
    "  filter(`sample == 0` == F)\n",
    "\n",
    "\n",
    "umendNonZerosDF <- dfNonZeros %>%\n",
    "  arrange(sampleID) %>%\n",
    "  filter(sampleID %in% umendResults$th_sampleid) %>%\n",
    "  add_column(rawUMEND = umendResults$umendCountRaw, umendID = umendResults$th_sampleid)\n",
    "\n",
    "ggplot(umendNonZerosDF, aes(n/1000, rawUMEND)) + geom_point() +\n",
    "  xlab(\"Thousands of Expressed Genes Per Sample\") + ylab(\"Raw UMEND Count\") +\n",
    "  geom_smooth(method = 'lm', level = 0) +\n",
    "  ggtitle(\"Number of Expressed Genes verses UMEND Reads Per Sample\") +\n",
    "  annotate(\n",
    "    \"text\",\n",
    "    x = 23.5,\n",
    "    y = 7e+07,\n",
    "    label = paste0(\n",
    "      \"correlation: \",\n",
    "      round(cor(umendNonZerosDF$rawUMEND,umendNonZerosDF$n),3)\n",
    "    )\n",
    "  )\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# ==================================================================================================\n",
    "# Correlation of 1\n",
    "# ==================================================================================================\n",
    "\n",
    "# REFERENCE\n",
    "x <- seq(1,5)\n",
    "y <- x\n",
    "df = data.frame(x,y)\n",
    "\n",
    "cor(x,y)\n",
    "ggplot(df, aes(x,y)) + geom_point() + ggtitle(\"Correlation of 1\")\n",
    "# correlation of 1 (when x = y)\n",
    "\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "3.3.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
